### PR TITLE
Set canonical location for downloads to artifacts.k8s.io

### PR DESCRIPTION
**Pull request by hakman**
And remove the legacy location for downloads.

/cc @johngmyers @rifelpet

### CODE REVIEW

**Comment by johngmyers** [comment]
Needs rebase.

**Comment by hakman** [comment]
Should be rebased now. Just that ci-robot is a little slow.

File: docs/release-process.md [hunk 1]
```diff
@@ -219,36 +215,22 @@ hub pull-request -b main
...
-crane cp gcr.io/k8s-staging-kops/kube-apiserver-healthcheck:${VERSION} kope/kube-apiserver-healthcheck:${VERSION}
-crane cp gcr.io/k8s-staging-kops/dns-controller:${VERSION} kope/dns-controller:${VERSION}
-crane cp gcr.io/k8s-staging-kops/kops-controller:${VERSION} kope/kops-controller:${VERSION}
+cd ${GOPATH}/src/k8s.io/kops/
+shipbot -tag v${VERSION} -config .shipbot.yaml -src ${GOPATH}/src/k8s.io/k8s.io/k8s-staging-kops/kops/releases/${VERSION}/
 ```
 
-
-Binaries to S3 bucket:
+Binaries to S3 bucket (only for kOps < 1.22):
```
Comment (johngmyers) [inline docs/release-process.md:228]: Do we do this for Beta releases?

File: docs/releases/1.22-NOTES.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ This is a document to gather the release notes prior to the release.
 
 * Support for Kubernetes version 1.17 is deprecated and will be removed in kOps 1.23.
 
+* The legacy location for downloads `s3://https://kubeupv2.s3.amazonaws.com/kops/` has been deprecated and will not be used for new releases. The new canonical downloads location is `https://artifacts.k8s.io/binaries/kops/`.
```
Comment (johngmyers) [inline docs/releases/1.22-NOTES.md:21]: Add to 1.21-NOTES.md. For 1.22-NOTES.md, put this in the "Breaking changes" section.

**Review by johngmyers — CHANGES_REQUESTED** [review]
It also causes the original URL to be used before mirrors instead of after. What is the reason for that?

File: docs/releases/1.22-NOTES.md (hunk 2 above)
Comment (hakman) [reply docs/releases/1.22-NOTES.md]: Done.

File: docs/release-process.md (hunk 1 above)
Comment (hakman) [reply docs/release-process.md]: Not quite related, but updated.

File: docs/release-process.md [hunk 3]
```diff
@@ -219,37 +215,23 @@ hub pull-request -b main
...
-Binaries to github:
-
-```
-cd ${GOPATH}/src/k8s.io/kops/
-shipbot -tag v${VERSION} -config .shipbot.yaml -src ${GOPATH}/src/k8s.io/k8s.io/k8s-staging-kops/kops/releases/${VERSION}/
-```
-
-
-Until the binary promoter is automatic, we also need to promote the binary artifacts manually:
+Until the binary promoter is automatic, we also need to promote the binary artifacts manually (only for releases):
```
Comment (johngmyers) [inline docs/release-process.md:234]:
```suggestion
Until the binary promoter is automatic, we also need to promote the binary artifacts manually (only for stable releases):
```

They're all "releases".

**Review by johngmyers — APPROVED** [review]

**Pull request merged**